packages/realm/src/Realm.ts

```diff
@@ -511,16 +511,19 @@ export class Realm {
       );
       const table = binding.Helpers.getTable(this.internal, objectSchema.tableKey);
       table.removeObject(obj.key);
-    } else if (Array.isArray(subject) || Symbol.iterator in subject) {
+    } else if (Array.isArray(subject) || subject instanceof Results) {
```

Comment:
I would add another branch above conditional on `subject instanceof Results` calling the `subject.clear()` (which needs to be added to the packages/bindgen/spec.yml first)

```suggestion
    } else if (Array.isArray(subject) || Symbol.iterator in subject) {
```

Reply:
This is done